### DISCUSSION TITLE
Wrap 2 Stack Screens with a Single Context Provider

### DISCUSSION THREAD

**Discussion by Kaung-Htet-Hein** [react-navigation/react-navigation#11644] [Q&A]
**Issue Title:**

Wrap Multiple Screens with a Single Context Provider

**Problem:**

I'm working on a React Navigation project, and I have a use case where I want to wrap 2 screens with a single context provider, but I don't want to wrap each screen individually. Currently, it seems like I have to wrap screens one by one, but I would like to achieve this with a single context provider wrapping 2 screens together.

**Expected Behavior:**

I would like to have the ability to create a custom component that wraps two or more screens together with a context provider, while not affecting other screens in the stack.

**Current Code:**

```jsx
<Stack.Navigator>
  <SomeContextProvider>
  <Stack.Screen
    name={ProductDetailScreen}
    component={ProductDetail}
  />
  <Stack.Screen
    name={ProductGalleryScreen}
    component={ProductGallery}
  />
  <SomeContextProvider/>

  <Stack.Screen
    name={OtherScreens}
    component={OtherScreen}
  />
  {/* ... other screens */}
</Stack.Navigator>
```

**Comment by raajnadar** [comment]
Could you please follow the below and let me know if that worked for you?

https://reactnavigation.org/docs/group/

**Reply by Kaung-Htet-Hein** [reply]
hi @raajnadar , it didn't work, brother.

**Comment by satya164** [comment]
Why do you need this? There's no harm in wrapping the whole navigator in a context, if a screen doesn't use the context then it's not affected by the provider.

To answer your question, it's not possible. The `Stack.Screen` are configuration objects, not actual components that are rendered.

**Reply by Kaung-Htet-Hein** [reply]
thanks bro @satya164 , i just wanted to know if that's possible, i just didn't want to wrap whole navigation container with a state provider wrapper. it makes it look messy

**Reply by raajnadar** [reply]
@satya164 one reason might be to prevent the unnecessary re-render of the screen that does not depend on that context

E.g. - We have 4 screen, Home, Shopping Cart, Profile, Settings, the shopping cart context is not required in the settings, so that the screen does not re-render on context change.

P.S. - Imagine the context at the root of the application and we have a lots of screen getting re rendered.

**Reply by satya164** [reply]
@raajnadar that's already how context works. screens that don't consume the context don't re-render regardless what's wrapped by it.

**Reply by raajnadar** [reply]
In the OPs question they only want to wrap specific screen under a context, probably they don't want the `OtherScreens` to get that context scope to prevent the re render.

**Reply by satya164** [reply]
As I wrote, they are not going to re-render if they don't consume the context. It makes no difference whether you wrap your whole app or few components.

**Reply by eric-spt** [reply]
@satya164 How about for use cases where the state should be reset when the involved screens are unmounted? (e.g a transfer flow where user select amount, recipient over multiple screens, and once the transfer flow is completed, all data is no longer relevant) With redux implementation I guess can either reset those global state upon starting the next transfer/upon the transfer flow completed, however those can be prone to human error. Interested to know your thought, thanks in advance!
